File: lib/core/utils/comp_order.dart
// Canonical display order for the competition families. Names are folded (alef /
// ya / ta-marbuta variants and diacritics) before matching, so a spelling
// variant still lines up. Anything not listed sorts after these, alphabetically.
// Mirrors web/src/lib/compOrder.ts.
const _canonical = [
  'دوري الجمهورية القسم الاول',
  'دوري الجمهورية القسم الثاني أ',
  'دوري الجمهورية القسم الثاني ب',
  'القطاعات',
  'المناطق',
];

final _diacritics = RegExp('[ً-ْـ]');
final _alef = RegExp('[أإآٱ]');
final _spaces = RegExp(r'\s+');

String _fold(String s) => s
    .toLowerCase()
    .replaceAll(_diacritics, '')
    .replaceAll(_alef, 'ا')
    .replaceAll('ى', 'ي')
    .replaceAll('ة', 'ه')
    .replaceAll(_spaces, ' ')
    .trim();

final List<String> _order = _canonical.map(_fold).toList();

int compRank(String name) {
  final i = _order.indexOf(_fold(name));
  return i == -1 ? _order.length : i;
}

/// Sort competition names by the canonical order, then alphabetically.
int compareCompName(String a, String b) {
  final r = compRank(a) - compRank(b);
  return r != 0 ? r : a.compareTo(b);
}

File: web/src/lib/compOrder.ts
// Canonical display order for the competition families. Names are folded (alef /
// ya / ta-marbuta variants and diacritics) before matching, so a spelling
// variant still lines up. Anything not listed sorts after these, alphabetically.
const CANONICAL = [
  'دوري الجمهورية القسم الاول',
  'دوري الجمهورية القسم الثاني أ',
  'دوري الجمهورية القسم الثاني ب',
  'القطاعات',
  'المناطق',
];

const fold = (s: string) =>
  (s || '')
    .toLowerCase()
    .replace(/[ً-ْـ]/g, '')
    .replace(/[أإآٱ]/g, 'ا')
    .replace(/ى/g, 'ي')
    .replace(/ة/g, 'ه')
    .replace(/\s+/g, ' ')
    .trim();

const ORDER = CANONICAL.map(fold);

export function compRank(name: string): number {
  const i = ORDER.indexOf(fold(name));
  return i === -1 ? ORDER.length : i;
}

/** Sort competition names by the canonical order, then alphabetically. */
export function compareCompName(a: string, b: string): number {
  return compRank(a) - compRank(b) || a.localeCompare(b, 'ar');
}
